python/lsst/afw/table/base/baseContinued.py

```diff
@@ -127,8 +127,18 @@ def __getitem__(self, key):
                 stop = len(self)
             return self.subset(start, stop, step)
         elif isinstance(key, np.ndarray):
+            if not self.isContiguous():
+                cat = self.__class__(self.table)
+                for rr, ok in zip(self, key):
```

Comment:
This block feels somewhat magical: can we have a comment explaining how it's supposed to work?